sources/commands/option_external.c

```diff
@@ -28,14 +33,33 @@ static char	*find_executable(char *tgt, t_dlist *path)
 	return (0);
 }
 
-static int	run_executable(char *fullpath, t_param *param)
+static int	run_executable(char *fullpath, t_treenode *node, t_param *param)
 {
-	char	*argv[2];
+	char	**argv;
+	int		argc;
+	int		i;
+	char	**envvars;
 
+	argc = node->children_qtty;
+	argv = ft_calloc_if(sizeof(char *) * (argc + 2), 1);
+	if (!argv)
+		return (ft_assign_i(&param->errno, 5, 0));
 	argv[0] = fullpath;
-	argv[1] = 0;
-	return (execve(fullpath, argv,
-			get_envvars_for_execve(param)));
+	argv[argc + 1] = 0;
+	i = 1;
+	node = node->child;
+	while (i <= argc)
+	{
+		argv[i ++] = node->content;
+		node = node->sibling_next;
+	}
+	envvars = get_envvars_for_execve(param);
```

Comment:
5.5 collecting environment variables into a line. getting PATH from a dlist, and all others from a mapss.